Client/WarFare/UIItemUpgrade.cpp

```diff
@@ -0,0 +1,1289 @@
+	{
+		pImg->SetVisible(true);
+		pImg->SetParent(this);
+	}
+
+}
+void CUIItemUpgrade::AnimClose()
+{
+	if (m_eAnimationState != ANIM_NONE)
+	{
+		float m_fAnimationTimer = 0.0f;
+		int m_iCurrentFrame = 0;
+		m_pImageCover1->SetVisible(false);
+		m_pImageCover2->SetVisible(false);
+	}
+
+	m_eAnimationState = ANIM_NONE;
+	HideAllAnimationFrames();
+}
+
+void CUIItemUpgrade::ShowResultUpgrade()
+{
+	// Show result item in result area if upgrade was successful
+	if (m_bUpgradeSuccesfull && m_pUpgradeResultSlot && m_pAreaResult)
+	{
+		// Set icon position to result area
+		m_pUpgradeResultSlot->pUIIcon->SetRegion(m_pAreaResult->GetRegion());
+		m_pUpgradeResultSlot->pUIIcon->SetMoveRect(m_pAreaResult->GetRegion());
+		m_pUpgradeResultSlot->pUIIcon->SetParent(this);
+		m_pUpgradeResultSlot->pUIIcon->SetVisible(true);
+	}
+
+	m_eAnimationState = ANIM_COVER_OPENING;
+}
+
+void CUIItemUpgrade::StartUpgradeAnim()
+{
+	if (m_bReceivedResultFromServer)
+	{
+		if (!m_pImageCover1 || !m_pImageCover2)
+			return;
+		m_fAnimationTimer = 0.0f;
+		m_iCurrentFrame = 0;
+
+		m_pImageCover1 = (CN3UIImage*) (this->GetChildByID("img_cover_01"));
+		__ASSERT(m_pImageCover1, "NULL UI Component!!");
+		m_pImageCover2 = (CN3UIImage*) (this->GetChildByID("img_cover_02"));
+		__ASSERT(m_pImageCover2, "NULL UI Component!!");
+
+		// Make covers visible during animation
+		m_pImageCover1->SetVisible(true);
+		m_pImageCover2->SetVisible(true);
+		m_pImageCover1->SetParent(this);
+		m_pImageCover2->SetParent(this);
+
+		// save original positions
+		m_rcCover1Original = m_pImageCover1->GetRegion();
+		m_rcCover2Original = m_pImageCover2->GetRegion();
+
+		m_eAnimationState = ANIM_FLIPFLOP;
+	}
+}
+
+void CUIItemUpgrade::HandleInventoryIconRightClick()
+{
+    // Get mouse position
+    POINT ptCur = CGameProcedure::s_pLocalInput->MouseGetPos();
+	uint32_t dwMouseFlags = CGameProcedure::s_pLocalInput->MouseGetFlag();
+    // Check for right mouse button click using MOUSE_RBCLICKED
+    if (dwMouseFlags & MOUSE_RBCLICKED)
+    {
+        // Find the icon under the mouse
+        for (int i = 0; i < MAX_ITEM_INVENTORY; ++i)
+        {
+            __IconItemSkill* pIconSkill = m_pMyUpgradeInv[i];
+            if (!pIconSkill || !pIconSkill->pUIIcon)
+				continue;
+
+            CN3UIIcon* pIcon = pIconSkill->pUIIcon;
+            const RECT& rc = pIcon->GetRegion();
+            if (pIcon->IsVisible() && rc.left <= ptCur.x && ptCur.x <= rc.right &&
+                rc.top <= ptCur.y && ptCur.y <= rc.bottom)
+            {
+                // Call HandleSlotDrop and HandleUpgradeAreaDrop for the icon under the mouse
+				if (HandleUpgradeAreaDrop(pIconSkill))
+				{
+					pIconSkill->pUIIcon->SetRegion(m_pAreaUpgrade->GetRegion());
+					pIconSkill->pUIIcon->SetMoveRect(m_pAreaUpgrade->GetRegion());
+					pIconSkill->pUIIcon->SetParent(this);
+				}
+				else
+				{
+					for (int i = 0; i < MAX_ITEM_UPGRADE_SLOT; ++i)
+					{
+						if (HandleSlotDrop(pIconSkill, i)) break;
```

Comment:
Move the `break;` to the next line, but at this point it might as well be a `return;`